components/browser/menu/src/main/java/mozilla/components/browser/menu/WebExtensionBrowserMenuBuilder.kt

```diff
@@ -88,28 +88,38 @@ class WebExtensionBrowserMenuBuilder(
             val webExtMenu = WebExtensionBrowserMenu(webExtBrowserMenuAdapter, store)
 
             ParentBrowserMenuItem(
-                label = context.getString(R.string.mozac_browser_menu_addons),
+                label = context.getString(R.string.mozac_browser_menu_addons2),
                 imageResource = R.drawable.mozac_ic_extensions,
                 iconTintColorResource = webExtIconTintColorResource,
                 subMenu = webExtMenu,
                 endOfMenuAlwaysVisible = endOfMenuAlwaysVisible
             )
         } else {
             BrowserMenuImageText(
-                label = context.getString(R.string.mozac_browser_menu_addons),
+                label = context.getString(R.string.mozac_browser_menu_addons2),
                 imageResource = R.drawable.mozac_ic_extensions,
                 iconTintColorResource = webExtIconTintColorResource
             ) {
                 onAddonsManagerTapped.invoke()
             }
         }
 
-        val menuItems =
-            if (appendExtensionSubMenuAtStart) {
+        val mainMenuIndex = finalList.indexOfFirst { browserMenuItem ->
+            (browserMenuItem as? WebExtensionPlaceholderMenuItem)?.id ==
+                WebExtensionPlaceholderMenuItem.MAIN_EXTENSIONS_MENU_ID
+        }
+
+        val menuItems = if (mainMenuIndex != -1) {
+            finalList[mainMenuIndex] = webExtMenuItem
+            finalList
+            // if we do not have a placeholder we should add the extension submenu at top or bottom
```

Comment:
If there is no placeholder then I think we should not show the extensions menu at all

Reply:
That was my initial thought too because Fenix would not need such a behavior, but I wanted to keep this option for other clients that might not need placement inside the list, as Fenix does now, for the moment. And it does not need any different parameters than we already use for placing the back button at the top or bottom. But I am good, either way if we decide to remove the fallback behavior.